[14.0][MIG] stock_valuation_layer_usage and stock_account_product_run_fifo_hook

ChrisOForgeFlow opened the pull request:
@ForgeFlow

Introduced new field to check original layer on outcome layers
Removed reservation case, because on core don't get any traceability data to assign candidate layer, only get by order and quantity available(field remaining_qty)

JordiBForgeFlow committed:
[13.0][add] stock_valuation_layer_usage, stock_account_product_run_fifo_hook

JordiBForgeFlow committed:
[13.0][add] stock_valuation_layer_usage, stock_account_product_run_fifo_hook

oca-travis committed:
[UPD] Update stock_account_product_run_fifo_hook.pot

oca-travis committed:
[UPD] Update stock_valuation_layer_usage.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

AaronHForgeFlow committed:
[13.0][IMP] stock_account_product_run_fifo_hook: add hook for layer selection on out svl creation

AaronHForgeFlow committed:
[13.0][IMP] stock_valuation_layer_usage: change output layer selection to respect mto case

AaronHForgeFlow committed:
[13.0][IMP]stock_account_product_run_fifo_hook: hook on condition to update standard price (#3)

OCA-git-bot committed:
stock_account_product_run_fifo_hook 13.0.1.1.0

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_valuation_layer_usage 13.0.1.1.0

ChrisOForgeFlow committed:
[IMP] stock_valuation_layer_usage: black, isort, prettier

ChrisOForgeFlow committed:
[IMP] stock_account_product_run_fifo_hook: black, isort, prettier

ChrisOForgeFlow committed:
[14.0][MIG] stock_valuation_layer_usage and stock_account_product_run_fifo_hook

AaronHForgeFlow reviewed (approved):
Functional review. Working fine. This is not WIP anymore, is it?

JordiBForgeFlow committed:
[FIX][stock_valuation_layer_usage]: look for true incoming moves
In case that the out move is connected with an internal transfer,
no valuation layer will be found, and then the system will create
a layer with negative quantity. We should try to look for the
true incoming move, if there is any.

AaronHForgeFlow committed:
[IMP] add some ux improvements

MiquelRForgeFlow commented on stock_account_product_run_fifo_hook/hooks.py:
```diff
@@ -18,12 +18,7 @@ def _run_fifo_new(self, quantity, company):
-        candidates = (
-            self.env["stock.valuation.layer"]
-            .sudo()
-            .with_context(active_test=False)
```
why?

MiquelRForgeFlow commented on stock_valuation_layer_usage/__manifest__.py:
```diff
@@ -3,15 +3,15 @@
     "author": "ForgeFlow,Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/stock-logistics-workflow",
-    "depends": ["stock_account_product_run_fifo_hook"],
+    "depends": ["sale", "stock_account_product_run_fifo_hook"],
```
why?

rousseldenis commented on stock_valuation_layer_usage/models/stock_valuation_layer.py:
```diff
@@ -29,11 +29,13 @@ class StockValuationLayer(models.Model):
+            lots = rec.stock_move_id.mapped("move_line_ids.lot_id").ids
+            if lots:
+                domain += [("stock_move_id.move_line_ids.lot_id", "in", lots)]
+            incoming_usages = self.env["stock.valuation.layer.usage"].search(domain)
```
@ChrisOForgeFlow Isn't there a mean to avoid this search in the loop ?

MiquelRForgeFlow reviewed (approved):
@ChrisOForgeFlow please, squash commits in a beautiful and meaningful way :)

MiquelRForgeFlow commented:
@rousseldenis @JordiBForgeFlow the PR is ready

sysadminmatmoz reviewed (approved):
It works for me... tested extensively in a big database with +100000s stock moves, abusing and simulating bad behaviour. Used a similar hook (identical in functionality) in production version 12 for 3 years.

AaronHForgeFlow commented:
ready to merge IMHO

rousseldenis commented:
/ocabot merge patch
